Filrb/Internal/Model.lean

```diff
@@ -169,27 +197,181 @@ theorem bst_iff_sorted_inorder {t : Raw α} : t.BST ↔ Sorted t.inorder := by
 namespace Raw
 namespace Model
 
-omit [Preorder α] [Ord α] [LawfulOrd α] in
+omit [Ord α] [LawfulOrd α] in
 @[simp]
-theorem inorder_paintColor_independent (t : Raw α) :
-    (t.paintColor c).inorder = t.inorder := by
+lemma inorder_paintColor_independent (t : Raw α ) : (t.paintColor c).inorder = t.inorder := by
   unfold paintColor
   split <;> simp
 
-theorem baliL_inorder_independent {l r : Raw α}
-    (hl1 : ∀ y ∈ l, y < x) (hl2 : BST l)
-    (hr1 : ∀ y ∈ r, x < y) (hr2 : BST r) :
-    (baliL x l r).inorder = l.inorder ++ x :: r.inorder := by
-  sorry
+omit [Ord α] [LawfulOrd α] in
+lemma baliL_inorder_independent{l r : Raw α} (hl1 : ∀ y ∈ l, y < x) (hl2 : BST l) (hr1 : ∀ y ∈ r, x < y) (hr2 : BST r)
+    : (baliL x l r).inorder = l.inorder ++ x :: r.inorder := by
```

Comment:
The mathlib style would want this theorem to be written like so:
```lean
lemma baliL_inorder_independent{l r : Raw α} (hl1 : ∀ y ∈ l, y < x) (hl2 : BST l)
    (hr1 : ∀ y ∈ r, x < y) (hr2 : BST r) :
    (baliL x l r).inorder = l.inorder ++ x :: r.inorder := by
```
note the two major differences here:
1. No leading `:` in the last line
2. Line length limit of 120 characters

Please make sure your theorems generally stick to this style when they become multi line due to the line limit, if they fit within 120 characters you can just leave them